File: bench/prime.lua
maxnum = 100000
i = 2

function is_prime(n) -- n is local implicitly
    local j = 2 -- make it local, performance critical
    while j * j < n + 1 do
        if n % j == 0 then
            return false
        end
        j = j + 1
    end
    return true
end

while i <= maxnum do
    if is_prime(i) then
        print(i)
    end
    i = i + 1
end

File: bench/prime.py
maxnum = 100000
i = 2

def is_prime(n):
    j = 2
    while j * j < n + 1:
        if n % j == 0:
            return False
        j += 1
    return True

while i <= maxnum:
    if is_prime(i):
        print(i)
    i += 1
